apps/theming-designer/src/components/SemanticSlots.tsx

```diff
@@ -432,20 +44,102 @@ export const SemanticSlots: React.StatelessComponent<ISemanticSlotsProps> = (pro
     );
   };
 
-  const fillVariantSlotsList = (variantType: VariantThemeType, mappingType: {}): JSX.Element[] => {
+  const trimPaletteColors = (paletteColors: {}): any => {
+    let trimmedPaletteColors = {};
+    for (let palette in paletteColors) {
+      if (
+        palette.includes('theme') ||
+        palette.includes('neutral') ||
+        palette.includes('black') ||
+        palette.includes('white') ||
+        palette.includes('redDark') ||
+        palette.includes('accent')
+      ) {
+        (trimmedPaletteColors as any)[palette] = (paletteColors as any)[palette];
+      }
+    }
+    return trimmedPaletteColors;
+  };
+
+  const trimSemanticColors = (semanticColors: {}): any => {
+    let trimmedSemanticColors = {};
+    for (let semanticColor in semanticColors) {
+      if (
+        semanticColor.includes('body') ||
+        semanticColor.includes('disabled') ||
+        semanticColor.includes('focus') ||
+        semanticColor.includes('variant') ||
+        semanticColor.includes('default') ||
+        semanticColor.includes('action') ||
+        semanticColor.includes('link') ||
+        semanticColor.includes('accent') ||
+        semanticColor.includes('list') ||
+        semanticColor.includes('button') ||
+        semanticColor.includes('menu') ||
+        semanticColor.includes('input')
+      ) {
+        (trimmedSemanticColors as any)[semanticColor] = (semanticColors as any)[semanticColor];
+      }
+    }
+    return trimmedSemanticColors;
+  };
+
+  const trimSemanticSlotNames = (semanticSlotNames: string[]): any => {
+    let trimmedSemanticSlotNames = [];
+    for (let i = 0; i < semanticSlotNames.length; i++) {
+      let slotName = (semanticSlotNames as any)[i];
+      if (
+        slotName.includes('body') ||
+        slotName.includes('disabled') ||
+        slotName.includes('focus') ||
+        slotName.includes('variant') ||
+        slotName.includes('default') ||
+        slotName.includes('action') ||
+        slotName.includes('link') ||
+        slotName.includes('accent') ||
+        slotName.includes('list') ||
+        slotName.includes('button') ||
+        slotName.includes('menu') ||
+        slotName.includes('input')
+      ) {
+        trimmedSemanticSlotNames.push(slotName);
+      }
+    }
+    return trimmedSemanticSlotNames;
+  };
+
+  const fillVariantSlotsList = (variantType: VariantThemeType): JSX.Element[] => {
     let currThemeVariant: ITheme;
     if (props.theme) {
       currThemeVariant = getVariant(props.theme, variantType);
       let currVariantSemanticColors = currThemeVariant.semanticColors;
+      const trimmedSemanticColors = trimSemanticColors(currVariantSemanticColors);
+      console.log('semantic colors', trimmedSemanticColors);
+      let currVariantPaletteColors = currThemeVariant.palette;
+      const onlyPaletteColors = trimPaletteColors(currVariantPaletteColors);
+      console.log('palette colors: ', onlyPaletteColors);
```

Comment:
I know this is a draft PR, but I just wanted to tag these for removal (assuming you don't mean to leave them in.)